src/auth.cpp

```diff
@@ -42,4 +47,65 @@ void PlainTextAuthenticator::on_authenticate_success(const std::string& token) {
   // no-op
 }
 
+SaslAuthenticator::SaslAuthenticator(const Address& host,
+                                     const CassAuthCallbacks* callbacks,
+                                     void* data)
+  : callbacks_(callbacks)
+  , data_(data) {
+  auth_.data = NULL;
+  auth_.host.address_length = host.to_inet(auth_.host.address);
+}
+
+SaslAuthenticator::~SaslAuthenticator() {
+  if (callbacks_->cleanup_callback != NULL) {
+    callbacks_->cleanup_callback(&auth_, data_);
+  }
+}
+
+std::string SaslAuthenticator::initial_response() {
+  if (callbacks_->initial_callback == NULL) {
+    return std::string();
+  }
+
+  std::string response(SASL_AUTH_INIT_RESPONSE_SIZE, '\0');
+  for (int i = 0; i < 2; ++i) {
+    size_t size = callbacks_->initial_callback(&auth_, data_,
+                                               &response[0], response.size());
```

Comment:
Just curious why ``&response[0]`` for string instead of ``response.data()`` in these calls.